src/machine_learning/connector.py

```diff
@@ -64,4 +65,11 @@ def generate_image(api_client: StableDiffusionAPI, text_prompt: str, negative_pr
     height = int(os.environ.get("HEIGHT", "512"))
 
     sd_api_client = StableDiffusionAPI(url)
-    generate_image(sd_api_client, text_prompt, disable_controlnet=disable_controlnet, width=width, height=height)
+    for _i in range(3):
```

Comment:
I would highly recommend against using time.sleep() instead, maybe listening to status updates and then triggering or using some non-blocking asynchronous methods

Reply:
The API is external to the code. I can hardly discover any alternative approach to navigate around it.